module/user-extension/helper.js

```diff
@@ -19,670 +18,806 @@ const programsHelper = require(MODULES_BASE_PATH + "/programs/helper");
-                    _id: {$in: programDocuments[0].components},
-                    programId: programId,
-                    isReusable: false,
-                    isDeleted: false
-                },[
-                    "externalId",
-                    "description",
-                    "name",
-                    "type",
-                    "subType",
-                    "isRubricDriven",
-                    "scoringSystem",
-                    "criteriaLevelReport"
-                ]);
-       
-                return resolve({
-                   message: constants.apiResponses.SOLUTIONS_LIST,
-                   data: solutionDocuments
-                });
-            } catch(error) {
-                return reject(error);
-            }
-        })
-    }
-};
-
-
+  static solutions(userId, programId, role) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const userInformation = await this.userExtensionDocument(
+          {
+            userId: userId,
+            status: constants.common.ACTIVE,
+            $and: [
+              {
+                platformRoles: {
+                  $elemMatch: { code: role, programs: ObjectId(programId) },
+                },
+              },
+            ],
+          },
+          {
+            _id: 1,
+          }
+        );
+
+        if (!userInformation) {
+          return resolve({
+            status: httpStatusCode.bad_request.status,
+            message: constants.apiResponses.USER_PLATFORM_ROLE_NOT_FOUND,
+          });
+        }
+
+        const programDocuments = await programsHelper.programDocuments(
+          {
+            _id: programId,
+            status: constants.common.ACTIVE,
+          },
+          ["components"]
+        );
+
+        if (!(programDocuments.length > 0)) {
+          return resolve({
+            status: httpStatusCode.bad_request.status,
+            message: constants.apiResponses.PROGRAM_NOT_FOUND,
+          });
+        }
+
+        const solutionDocuments = await solutionsHelper.solutionDocuments(
+          {
+            _id: { $in: programDocuments[0].components },
+            programId: programId,
+            isReusable: false,
+            isDeleted: false,
+          },
+          [
+            "externalId",
+            "description",
+            "name",
+            "type",
+            "subType",
+            "isRubricDriven",
+            "scoringSystem",
+            "criteriaLevelReport",
+          ]
+        );
+
+        return resolve({
+          message: constants.apiResponses.SOLUTIONS_LIST,
+          data: solutionDocuments,
+        });
+      } catch (error) {
+        return reject(error);
+      }
+    });
+  }
+
+  /**
+   * bulkWrite the userExtension collection
+   * @method
+   * @name updateMany
+   * @param {Array} query - Array of queries to update.
+   * @returns {Array}
+   */
 
+  static updateMany(query) {
```

Comment:
@praveenKDass As discussed keep the updateMany signature same as other modules. Also variable name doesn't make sense.